inginious/frontend/plugins/upcoming_tasks/__init__.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INGInious. See the LICENSE and the COPYRIGHTS files for
+# more information about the licensing of this file.
+#
+
+""" A plugin that allow students to see their futur work in a single place for all their courses """
+import os
+from collections import OrderedDict
+from datetime import datetime, timedelta
+import flask
+
+
+from flask import send_from_directory
+from inginious.frontend.pages.utils import INGIniousPage, INGIniousAuthPage
+
+PATH_TO_PLUGIN = os.path.abspath(os.path.dirname(__file__))
+
+def menu(template_helper):
+    """ Displays the link to the board on the main page, if the plugin is activated """
+    return template_helper.render("main_menu.html", template_folder=PATH_TO_PLUGIN + '/templates/')
+
+class StaticMockPage(INGIniousPage):
+    """ MockPage based on auto-evaluation plugin structure """
+
+    def GET(self, path):
+        return send_from_directory(os.path.join(PATH_TO_PLUGIN, "static"), path)
+
+    def POST(self, path):
+        return self.GET(path)
+
+
+class UpComingTasksBoard(INGIniousAuthPage):
+
+    def GET_AUTH(self):
+        """ Called when reaching the page """
+        time_planner = "unlimited"
+        return self.page(time_planner)
+
+    def POST_AUTH(self):
+        """ Called when modifying time planner """
+        user_input = flask.request.form
+        time_planner = "unlimited"
```

Comment:
Actually I noticed, simply using time_planner = user_input.get("time_planner") is enough since it's in the POST and we only enter here if the user opens the time_planer modal then validates it, so there is 100% always a selected value (the cancel button does not proc the POST).
This removes these 3 lines:

         time_planner = "unlimited"
         if "time_planner" in user_input:
               time_planner = user_input.get("time_planner", "unlimited")

into just: 

     time_planner = user_input.get("time_planner")

Reply:
You can still receive badly formatted or malicious requests, so you must have a fallback for all unexpected inputs.

Reply:
oh, ok. I didn't think about requests not coming from the user interface actually.
From the line you suggested, I suppose you meant the "unlimited" to act as a default value but I'm a bit surprised that the get method from the ImmutableMultiDict (the form) acts like that.